### DISCUSSION TITLE
Is there zephyr driver for ST LSM6DSR or can I use the LSM6DSO one for LSM6DSR?

### DISCUSSION THREAD

**Discussion by dehaoji** [zephyrproject-rtos/zephyr#65100] [Q&A] [answered]
I am using LSM6DSR with nrf 52811 using NCS +Zephyr. I saw there is driver and sample code for LSM6DSO. Is there one for LSM6DSR or can I just use the LSM6DSO driver for LSM6DSR?

Thanks,

**Comment by FRASTM** [comment, marked as answer]
STMicroelectronics has both:

- https://github.com/STMicroelectronics/lsm6dsr-pid
- https://github.com/STMicroelectronics/lsm6dso-pid

Look at  hal_st/sensor/stmemsc  repository:

- https://github.com/zephyrproject-rtos/hal_st/tree/master/sensor/stmemsc/lsm6dsr_STdC/driver
- https://github.com/zephyrproject-rtos/hal_st/tree/master/sensor/stmemsc/lsm6dso_STdC/driver

**Reply by dehaoji** [reply]
Thanks. I did see the driver code under hal folder. what I was asking is the driver code under zephyr/drivers/sensor folder, which has lsm6dso related code calling the driver code under hal folder. Nevermind, I have made one by referring to the lsm6dso one.
